library/Vanilla/AliasLoader.php

```diff
@@ -0,0 +1,28 @@
+<?php
+/**
+ * @author Adam Charron <adam.c@vanillaforums.com>
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+namespace Vanilla;
+
+/**
+ * An autoloader for vanilla class aliases.
+ *
+ * We need this because declaring a class_alias autoloads the class.
+ * Declaring aliases up front (like in the bootstrap) would autoload all of our classes.
+ * This class provides an autoloader for usage with spl_autoload_register to autoload these aliases
+ * which will then autoload their new classes if they are not loaded yet.
+ */
+class AliasLoader {
+
+    use AliasProviderTrait;
+
+    /**
+     * @inheritdoc
+     */
+    protected static function provideAliases(): array {
```

Comment:
It looks like this is intentionally hardcoded. Is the idea that only core Vanilla will be able to add class aliases and that those aliases will require manually updating the value returned from `AliasLoader::provideAliases`?

Reply:
That is correct.

This is the `Vanilla\AliasLoader` which is aliases for Vanilla. If another one of our projects, or someone else wants to provide aliases in a similar way the can use the `AliasProviderTrait` and register their own autoloader.

That is what the `TestAliasLoader` does as well.

Reply:
I am *slightly* concerned that other projects might not be able to register their autoloader early enough, but I don't think there's a high enough probability of that occurring to re-engineer anything here.